components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { hexToRgba } from '../shared';
+import { colors } from '../shared/theme';
+
+const Wrapper = styled.div`
+  animation: fadeIn ease-out 0.3s;
+  background-color: ${({
+    theme: {
+      colors: { neutral },
+    },
+  }) => hexToRgba(neutral[700], 0.5)};
+  display: flex;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+
+  z-index: 99;
+
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`;
+
+const Dialog = ({
+  onClose,
+  closeOnTime,
+  children,
+  theme,
+  disableClickOutside,
+  selector,
+}) => {
+  const dialogOver = useRef();
+
+  const [mounted, setMounted] = useState(false);
+
+  let timeout = null;
+
+  const handleClose = () => {
+    onClose();
+    clearTimeout(timeout);
+  };
+
+  const handleEscKey = ({ key }) => {
+    if (key === 'Escape') {
+      handleClose();
+    }
+  };
+
+  const handleClickOutside = ({ target }) => {
+    if (target === dialogOver.current && !disableClickOutside) {
+      handleClose();
+    }
+  };
+
+  if (closeOnTime && mounted) {
+    timeout = setTimeout(() => {
+      onClose();
+    }, closeOnTime * 1000);
+  }
+
+  useEffect(() => {
+    setMounted(true);
+    window.addEventListener('keydown', handleEscKey);
+    document.body.style.overflow = 'hidden';
+
+    return () => {
+      setMounted(false);
+      document.body.style.overflow = 'auto';
+      window.removeEventListener('keydown', handleEscKey);
+    };
+  }, []);
+
+  return mounted
+    ? createPortal(
+        <Wrapper
+          ref={dialogOver}
+          onClick={e => handleClickOutside(e)}
+          role="dialog"
+          tabIndex="-1"
+          theme={theme}
+        >
+          {children}
+        </Wrapper>,
+        document.querySelector(selector),
```

Comment:
i didnt understand it.
Why just create a section or div to renders the Dialog inside? Why be dinamic?

Reply:
You mean the querySelector or the ternary returning null? If is about the ternary, it's because that way it will always return null during initial render both on server and client side, that way it's better for next users. I'm thinking about adding the selector prop to the docs, that way might help next users if they want to iniciate the page with dialog open.

Reply:
I didn't understand too